### ISSUE TITLE
QuickCheck does not build on default Travis Haskell infrastructure

### ISSUE THREAD

**Issue by sol** [nick8325/quickcheck#154]
I suspect this is a combination of an outdated version of `cabal-install` and the dependency bloat that was added to `QuickCheck` over the last couple of months (`semigroups` @phadej).

Steps to reproduce are left as an exercise to the inclined reader.

**Comment by sol** [comment]
https://github.com/hspec/hspec-example/pull/10

**Comment by phadej** [comment]
I cannot comment exactly about `cabal-install-1.18.0.2`. Looks like it solver constructs non-buildable plan. At least the fact it picks `bytestring-builder-0.10.4.0` is not good.

I'm not sure where this should be fixed (in `semigroups`, or `bytestring-builder` bounds, or both). It's simpler to use newer `cabal-install`. If you want this to be resolved, please open an issue to https://github.com/haskell-infra/hackage-trustees/issues

TL;DR it's not QuickCheck's fault.

**Comment by sol** [comment]
> TL;DR it's not QuickCheck's fault.

Thats an interesting way to look at it.  But for me it's like I write software so that people can use it.  Specifically, I write `hspec` so that people can test software with it.  That `hspec` builds on Travis is part of that.  If I change something and that leads to `hspec` not building on Travis I track it down and try to fix it.  If I don't do that people will open issues.

**Comment by sol** [comment]
See https://github.com/haskell-infra/hackage-trustees/issues/91
